src/dataset/mod.rs

```diff
@@ -70,6 +70,18 @@ where
     weights: Vec<f32>,
 }
 
+/// Dataset
+///
+/// The most commonly used typed of dataset. It contains a number of records
+/// stored as an `Array2` and each record corresponds to a single target. Such
+/// targets are stored as an `Array1`.
+pub type Dataset<D, T> = DatasetBase<ArrayBase<OwnedRepr<D>, Ix2>, ArrayBase<OwnedRepr<T>, Ix1>>;
```

Comment:
should we have a `type DatasetPr<D> = DatasetBase<ArrayBase<OwnedRepr<D>, Ix2>, ArrayBase<OwnedRepr<Pr>, Ix2>>` as well to multiclass probabilities :thinking:

Reply:
I think that would be a handy shortcut